greatestPalindrome.js

```diff
@@ -0,0 +1,57 @@
+
+/*Given a string return the longest palindrome that can be constructed by removing or shuffling characters. 
+Example: 
+'aha' -> 'aha' 
+'ttaatta' -> ' ttaaatt' 
+'abc' -> 'a' or 'b' or 'c' 
+'gggaaa' -> 'gaaag' or 'aggga' 
+Note if there are multiple correct answers you only need to return 1 palindrome.
+*/
+
+var word = "aabczRc";
+var wordDict = {};
+var leftPalindrome = [];
+var rigthPalindrome = [];
+var palindrome;
+
+function dictionaryCreator(word){
+  for(var index in word){
+    var letter = word[index];
+    if(wordDict[letter]){
+      wordDict[letter]++;
+    }
+    else{
+      wordDict[letter] = 1;
+    }
+  }
+};
+
+function palindromeConstructor (wordDict){
+	for(var index in wordDict) {
+		for(var i=wordDict[index]; i>1;i-=2){
```

Comment:
you can simplify this from On^2 to On, you just
   go through the wordDict
      if there is a spare letter just
           divide between 2 the quantity and then get the even value and divide it by 2 and put the respective number between lef and right array. left = new +left;  right  = right + left;
      else there is no space
            divide by 2 if there is a spare number put it on spareletter
             then do as avove.
          at the end just do left+spare + right
and that'll make it a palidrome

no need to reversing (which is itself a  On^2,)